### DISCUSSION TITLE
How can i start up bleep single?

### DISCUSSION THREAD

**Discussion by tyloafer** [BloopAI/bloop#1260] [Q&A] [answered]
I startup by the follow steps
1.  `npm run start-web ` to start web
2. `/usr/local/bin/qdrant` to start qdrant
3. `run --package bleep  --bin bleep  -- --cognito-auth-url https://production-github-oauth.auth.eu-west-1.amazoncognito.com/oauth2/authorize --cognito-mgmt-url  http://localhost:5173/ --cognito-client-id 6iso6185sujqtfttt1ik63sctc` to start bleep
 
however, when i finish those steps, i get github login page and i cannot login into bloop?

how can i skip login or whant params should i give?

**Comment by anastasiya1155** [comment, marked as answer]
Hello @tyloafer, you can refer to the instructions in our [README.md](https://github.com/BloopAI/bloop/blob/oss/README.md#building-from-source), it also references another [README.md file in apps/desktop](https://github.com/BloopAI/bloop/blob/oss/apps/desktop/README.md). You need to read both of them to successfully run the app. Do note that you don't have to run qdrant or frontend separately, `npm run start-app` should start them automatically.

**Reply by tyloafer** [reply]
thanks

now i start up successfully

and i find that json should be enabled in `server/bleep/Cargo.toml` like
`reqwest = { version = "0.11.20", features = ["rustls-tls-webpki-roots", "cookies", "gzip", "json"], default-features = false }`

or will get error when compile

**Reply by anastasiya1155** [reply]
weird, should compile as is, what version of cargo and rust are you using?

**Reply by tyloafer** [reply]
when i use `npm run start-app`  it compile sucessfully with no modify

when i compile by `run --package bleep  --bin bleep  -- --config-file=/path/to/local_config.json` in rustrover, the error occurs

the version of cargo is 
cargo 1.77.2 (e52e36006 2024-03-26)
